examples/keras_recipes/creating_tfrecords.py

```diff
@@ -0,0 +1,291 @@
+        "id": tf.io.FixedLenFeature([], tf.int64),
+        "image_id": tf.io.FixedLenFeature([], tf.int64),
+    }
+    example = tf.io.parse_single_example(example, feature_description)
+    example["image"] = tf.io.decode_jpeg(example["image"], channels=3)
+    example["bbox"] = tf.sparse.to_dense(example["bbox"])
+
+    return example
+
+
+"""
+## Generating TFRecords
+
+The generated TFRecrods will have the file name format `file_{number}.tfrec`, this is
+optional, but writing the number of files at the file name can make counting easier.
+"""
+
+for tfrec_num in range(n_tfrecods):
+    samples = annotations[(tfrec_num * n_samples) : ((tfrec_num + 1) * n_samples)]
+
+    with tf.io.TFRecordWriter(
+        tfrecords_dir + "/file_%.2i-%i.tfrec" % (tfrec_num, len(samples))
+    ) as writer:
+        for sample in samples:
+            image_path = f"{images_dir}/{sample['image_id']:012d}.jpg"
+            image = tf.io.decode_jpeg(tf.io.read_file(image_path))
+            example = create_example(image, image_path, sample)
+            writer.write(example.SerializeToString())
+
+"""
+## Looking at a generated TFRecord of a single data sample
+"""
+
+raw_dataset = tf.data.TFRecordDataset(f"{tfrecords_dir}/file_00-{n_samples}.tfrec")
+parsed_dataset = raw_dataset.map(parse_tfrecord_fn)
+
+for features in parsed_dataset.take(1):
+    for key in features.keys():
+        if key != "image":
+            print(f"{key}: {features[key]}")
+
+    print(f"Image shape: {features['image'].shape}")
+    plt.figure(figsize=(7, 7))
+    plt.imshow(features["image"].numpy())
+    plt.show()
+
+"""
+## Training a simple model with the generated TFRecords
+
+Explain that you are able to do not use all features from the dataset
+"""
+
+"""
+## Dataset functions
+"""
+
+
+def prepare_sample(features):
+    image = tf.image.resize(features["image"], size=(224, 224))
+    return image, features["category_id"]
+
+
+def get_dataset(filenames, batch_size):
+    """
+        Return a Tensorflow dataset ready for training or inference.
+    """
+    dataset = (
+        tf.data.TFRecordDataset(
+            filenames, num_parallel_reads=tf.data.experimental.AUTOTUNE
+        )
+        .map(parse_tfrecord_fn, num_parallel_calls=tf.data.experimental.AUTOTUNE)
+        .map(prepare_sample, num_parallel_calls=tf.data.experimental.AUTOTUNE)
+        .shuffle(batch_size * 10)
+        .batch(batch_size)
+        .prefetch(tf.data.experimental.AUTOTUNE)
+    )
+    return dataset
+
+
+train_filenames = tf.io.gfile.glob(tfrecords_dir + "/*.tfrec")
+batch_size = 64
+epochs = 5
+steps_per_epoch = 50
+
+
+input_tensor = tf.keras.layers.Input(shape=(224, 224, 3), name="image")
+model = tf.keras.applications.EfficientNetB0(
+    input_tensor=input_tensor, weights=None, classes=91
+)
+
+
+model.compile(
+    optimizer=tf.keras.optimizers.Adam(),
+    loss=tf.keras.losses.SparseCategoricalCrossentropy(),
+    metrics=["sparse_categorical_accuracy"],
```

Comment:
Since you're passing the optimizer and loss as objects, for consistency also pass the metrics as objects